hackett-lib/hackett/private/toplevel.rkt

```diff
@@ -82,8 +83,17 @@
       (syntax/loc this-syntax
         (begin form ... (@%top-interaction . form*)))]
      [expr
-      (and~>> (get-type #'expr)
-              apply-current-subst
-              τ->string
-              (printf ": ~a\n"))
```

Comment:
the fact that the bug was caused by a side-effect-based implementation is strangely satisfying :)